In [2]:
import pandas as pd
from sklearn import datasets
import statsmodels.api as sm
from stargazer.stargazer import Stargazer

diabetes = datasets.load_diabetes()
df = pd.DataFrame(diabetes.data)
df.columns = ['Age', 'Sex', 'BMI', 'ABP', 'S1', 'S2', 'S3', 'S4', 'S5', 'S6']
df['target'] = diabetes.target

est = sm.OLS(endog=df['target'], exog=sm.add_constant(df[df.columns[0:4]])).fit()
est2 = sm.OLS(endog=df['target'], exog=sm.add_constant(df[df.columns[0:6]])).fit()


stargazer = Stargazer([est, est2])

In [3]:
stargazer.render_html()

'<table style="text-align:center"><tr><td colspan="3" style="border-bottom: 1px solid black"></td></tr><tr><td style="text-align:left"></td><td colspan="2"><em>Dependent variable:target</em></td></tr><tr><td style="text-align:left"></td><tr><td style="text-align:left"></td><td>(1)</td><td>(2)</td></tr><tr><td colspan="3" style="border-bottom: 1px solid black"></td></tr><tr><td style="text-align:left">ABP</td><td>416.674<sup>***</sup></td><td>397.583<sup>***</sup></td></tr><tr><td style="text-align:left"></td><td>(69.495)</td><td>(70.870)</td></tr><tr><td style="text-align:left">Age</td><td>37.241<sup></sup></td><td>24.704<sup></sup></td></tr><tr><td style="text-align:left"></td><td>(64.117)</td><td>(65.411)</td></tr><tr><td style="text-align:left">BMI</td><td>787.179<sup>***</sup></td><td>789.742<sup>***</sup></td></tr><tr><td style="text-align:left"></td><td>(65.424)</td><td>(66.887)</td></tr><tr><td style="text-align:left">S1</td><td></td><td>197.852<sup></sup></td></tr><tr><td style

In [4]:
stargazer.render_latex()

'\\begin{table}[!htbp] \\centering\n\\begin{tabular}{@{\\extracolsep{5pt}}lcc}\n\\\\[-1.8ex]\\hline\n\\hline \\\\[-1.8ex]\n& \\multicolumn{2}{c}{\\textit{Dependent variable:}} \\\n\\cr \\cline{2-3}\n\\\\[-1.8ex] & (1) & (2) \\\\\n\\hline \\\\[-1.8ex]\n ABP & 416.674$^{***}$ & 397.583$^{***}$ \\\\\n  & (69.495) & (70.870) \\\\\n Age & 37.241$^{}$ & 24.704$^{}$ \\\\\n  & (64.117) & (65.411) \\\\\n BMI & 787.179$^{***}$ & 789.742$^{***}$ \\\\\n  & (65.424) & (66.887) \\\\\n S1 & & 197.852$^{}$ \\\\\n  & & (143.812) \\\\\n S2 & & -169.251$^{}$ \\\\\n  & & (142.744) \\\\\n Sex & -106.578$^{*}$ & -82.862$^{}$ \\\\\n  & (62.125) & (64.851) \\\\\n const & 152.133$^{***}$ & 152.133$^{***}$ \\\\\n  & (2.853) & (2.853) \\\\\n\\hline \\\\[-1.8ex]\n Observations & 442 & 442 \\\\\n $R^2$ & 0.400 & 0.403 \\\\\n Adjusted $R^2$ & 0.395 & 0.395 \\\\\n Residual Std. Error & 59.976(df = 437) & 59.982(df = 435)  \\\\\n F Statistic & 72.913$^{***}$ (df = 4.0; 437.0) & 48.915$^{***}$ (df = 6.0; 435.0) \\\\\n


\\begin{table}[!htbp] \\centering\n\\begin{tabular}{@{\\extracolsep{5pt}}lcc}\n\\\\[-1.8ex]\\hline\n\\hline \\\\[-1.8ex]\n& \\multicolumn{2}{c}{\\textit{Dependent variable:}} \\\n\\cr \\cline{2-3}\n\\\\[-1.8ex] & (1) & (2) \\\\\n\\hline \\\\[-1.8ex]\n ABP & 416.674$^{***}$ & 397.583$^{***}$ \\\\\n  & (69.495) & (70.870) \\\\\n Age & 37.241$^{}$ & 24.704$^{}$ \\\\\n  & (64.117) & (65.411) \\\\\n BMI & 787.179$^{***}$ & 789.742$^{***}$ \\\\\n  & (65.424) & (66.887) \\\\\n S1 & & 197.852$^{}$ \\\\\n  & & (143.812) \\\\\n S2 & & -169.251$^{}$ \\\\\n  & & (142.744) \\\\\n Sex & -106.578$^{*}$ & -82.862$^{}$ \\\\\n  & (62.125) & (64.851) \\\\\n const & 152.133$^{***}$ & 152.133$^{***}$ \\\\\n  & (2.853) & (2.853) \\\\\n\\hline \\\\[-1.8ex]\n Observations & 442 & 442 \\\\\n $R^2$ & 0.400 & 0.403 \\\\\n Adjusted $R^2$ & 0.395 & 0.395 \\\\\n Residual Std. Error & 59.976(df = 437) & 59.982(df = 435)  \\\\\n F Statistic & 72.913$^{***}$ (df = 4.0; 437.0) & 48.915$^{***}$ (df = 6.0; 435.0) \\\\\n\\hline\n\\hline \\\\[-1.8ex]\n\\textit{Note:} & \\multicolumn{2}{r}{$^{*}$p$<$0.1; $^{**}$p$<$0.05; $^{***}$p$<$0.01} \\\\\n\\end{tabular}\n\\end{table}
